### ISSUE TITLE
Querying `true == All(...)`  is failed with MySqlException

### ISSUE THREAD

**Issue by nenono** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#1387]
### Steps to reproduce
Run query likes `.Where(x => true == x.SomeRelation.All(y => ...))`.

```
            var results = _context.Set<User>()
                .Where(x =>
                     true == x.UsersRoles.All(y => y.Role.Name == "Staff"))
            .ToList();
```

`_context` is a `Microsoft.EntityFrameworkCore.DbContext`. Tables are created by `Microsoft.AspNetCore.Identity`.

In my usecase, a boolean parameter replaces the `true` literal.

### The issue
The query is failed with MySqlException.

message: 

```
MySqlException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'NOT EXISTS (
SELECT 1
FROM `AspNetUserRoles` AS `a0`
INNER JOIN `' at line 3
```

stacktrace:

```
An unhandled exception has occurred while executing the request.
MySql.Data.MySqlClient.MySqlException (0x80004005): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'NOT EXISTS (
    SELECT 1
    FROM `AspNetUserRoles` AS `a0`
    INNER JOIN `' at line 3
 ---> MySql.Data.MySqlClient.MySqlException (0x80004005): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'NOT EXISTS (
    SELECT 1
    FROM `AspNetUserRoles` AS `a0`
    INNER JOIN `' at line 3
   at MySqlConnector.Core.ResultSet.ReadResultSetHeaderAsync(IOBehavior ioBehavior) in /_/src/MySqlConnector/Core/ResultSet.cs:line 51
   at MySql.Data.MySqlClient.MySqlDataReader.ActivateResultSet() in /_/src/MySqlConnector/MySql.Data.MySqlClient/MySqlDataReader.cs:line 130
   at MySql.Data.MySqlClient.MySqlDataReader.CreateAsync(CommandListPosition commandListPosition, ICommandPayloadCreator payloadCreator, IDictionary`2 cachedProcedures, IMySqlCommand command, CommandBehavior behavior, IOBehavior ioBehavior, CancellationToken cancellationToken) in /_/src/MySqlConnector/MySql.Data.MySqlClient/MySqlDataReader.cs:line 391
   at MySqlConnector.Core.CommandExecutor.ExecuteReaderAsync(IReadOnlyList`1 commands, ICommandPayloadCreator payloadCreator, CommandBehavior behavior, IOBehavior ioBehavior, CancellationToken cancellationToken) in /_/src/MySqlConnector/Core/CommandExecutor.cs:line 62
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteDbDataReader(CommandBehavior behavior) in /_/src/MySqlConnector/MySql.Data.MySqlClient/MySqlCommand.cs:line 210
   at System.Data.Common.DbCommand.ExecuteReader()
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReader(RelationalCommandParameterObject parameterObject)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.InitializeReader(DbContext _, Boolean result)
   at Microsoft.EntityFrameworkCore.Storage.ExecutionStrategy.ExecuteImplementation[TState,TResult](Func`3 operation, Func`3 verifySucceeded, TState state)
   at Microsoft.EntityFrameworkCore.Storage.ExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
(Here was some my code)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Logged|12_1(ControllerActionInvoker invoker)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.StatusCodePagesMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

output SQL from log:

```
[12:55:20 ERR] Failed executing DbCommand (4ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
SELECT `a`.`Id`, `a`.`AccessFailedCount`, `a`.`CanReceiveSms`, `a`.`ConcurrencyStamp`, `a`.`Email`, `a`.`EmailConfirmed`, `a`.`IsSkippedSmsAuth`, `a`.`LockoutEnabled`, `a`.`LockoutEnd`, `a`.`NormalizedEmail`, `a`.`NormalizedUserName`, `a`.`PasswordHash`, `a`.`PhoneNumber`, `a`.`PhoneNumberConfirmed`, `a`.`SecurityStamp`, `a`.`TwoFactorEnabled`, `a`.`UserName`
FROM `AspNetUsers` AS `a`
WHERE TRUE = NOT EXISTS (
    SELECT 1
    FROM `AspNetUserRoles` AS `a0`
    INNER JOIN `AspNetRoles` AS `a1` ON `a0`.`RoleId` = `a1`.`Id`
    WHERE (`a`.`Id` = `a0`.`UserId`) AND ((`a1`.`Name` <> 'Staff') OR `a1`.`Name` IS NULL))
```

Some columns are added for my project. Please ignore.

### Further technical details

MySQL version: 8.0.18
Operating system: Windows 10
Pomelo.EntityFrameworkCore.MySql version: 3.1.0, and I tried 3.2.4.
Microsoft.AspNetCore.App version: 2.2.8

TargetFramework: netcoreapp3.1
I build an application with AspNetCore.

### Workaround 
Rewrite the query like this.

```
            var results = _context.Set<User>()
                .Where(x =>
                     x.UsersRoles.All(y => y.Role.Name == "Staff") == true)
            .ToList();
```

This is output SQL.

```
[13:11:37 INF] Executed DbCommand (1ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
SELECT `a`.`Id`, `a`.`AccessFailedCount`, `a`.`CanReceiveSms`, `a`.`ConcurrencyStamp`, `a`.`Email`, `a`.`EmailConfirmed`, `a`.`IsSkippedSmsAuth`, `a`.`LockoutEnabled`, `a`.`LockoutEnd`, `a`.`NormalizedEmail`, `a`.`NormalizedUserName`, `a`.`PasswordHash`, `a`.`PhoneNumber`, `a`.`PhoneNumberConfirmed`, `a`.`SecurityStamp`, `a`.`TwoFactorEnabled`, `a`.`UserName`
FROM `AspNetUsers` AS `a`
WHERE NOT EXISTS (
    SELECT 1
    FROM `AspNetUserRoles` AS `a0`
    INNER JOIN `AspNetRoles` AS `a1` ON `a0`.`RoleId` = `a1`.`Id`
    WHERE (`a`.`Id` = `a0`.`UserId`) AND ((`a1`.`Name` <> 'Staff') OR `a1`.`Name` IS NULL)) = TRUE
```

**Comment by lauxjpn** [comment]
@nenono Thanks for reporting the issue.

This issue has already been fixed for EF Core 5 (and Pomelo 5).

Since there the expression work with a trailing (instead of a leading) or no explicit `true` comparison at all (which is the most common syntax), I don't think we are going to fix this for EF Core 3.1 (our 3.2 branch) anymore.

**Comment by nenono** [comment]
@lauxjpn

> This has already been fixed for EF Core 5 (and Pomelo 5).

Oh, my project is not upgraded yet. I use workaround above.

> Since the expression works with a trailing (instead of a leading) or no explicit true comparison at all (which is the most common syntax), I don't think we are going to fix this for EF Core 3.1 (our 3.2 branch) anymore.

👍 

I was in trouble when using a input boolean parameter. 

```
            var inputValue = ...;
            var results = _context.Set<User>()
                .Where(x =>
                     inputValue == x.UsersRoles.All(y => y.Role.Name == "Staff"))
            .ToList();
```

But I think it is a small issue for supports old versions.

Thank you.
